### PR TITLE
Add questionnaire deletion confirmation

### PR DESCRIPTION
**Pull request by rmccar**
### What is the context of this PR?
Added confirmation modal for when a questionnaire is deleted.

When a questionnaire is deleted the animation isn't the smoothest and might need to be looked at in future

### How to review
Create and delete a questionnaire to make sure modal is shown and works correctly.

### CODE REVIEW

**Comment by rmccar** [comment]
[Add questionnaire deletion confirmation](https://trello.com/c/IKz0lNtD/919-add-questionnaire-deletion-confirmation)

File: eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.js [hunk 1]
```diff
@@ -93,47 +115,60 @@ class Row extends React.Component {
   render() {
     const { questionnaire, onDeleteQuestionnaire, ...rest } = this.props;
     const isOptimisticDupe = this.isQuestionnaireADuplicate();
+    const { showDeleteQuestionnaireDialog } = this.state;
+    console.log(questionnaire.displayName, onDeleteQuestionnaire, rest);
```
Comment (SamGodwin2) [inline eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.js]: Did this get left in by accident?

File: eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.js [hunk 2]
```diff
@@ -93,47 +115,60 @@ class Row extends React.Component {
...
-                hideText
-                data-test="btn-delete-questionnaire"
-                onClick={partial(onDeleteQuestionnaire, questionnaire.id)}
+      <>
+        <DeleteConfirmDialog
+          isOpen={showDeleteQuestionnaireDialog}
+          onClose={this.handleCloseDeleteQuestionnaireDialog}
+          onDelete={this.handleDeleteQuestionnaire}
+          title={questionnaire.title}
+          alertText="Questionnaire will be deleted."
```
Comment (SamGodwin2) [inline eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.js]: Do we want to also add that all sections and pages will be deleted to have it consistent with the section delete modal?

File: eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.js [hunk 3]
```diff
@@ -93,47 +115,60 @@ class Row extends React.Component {
...
+          icon={questionConfirmationIcon}
```
Comment (SamGodwin2) [inline eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.js]: screenshot 2019-02-06 at 15 06 31

Icon looks a little small?

Comment (rmccar) [reply eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.js]: This is also now sorted with a new icon

File: eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.js [hunk 4]
```diff
@@ -91,49 +116,61 @@ class Row extends React.Component {
...
-                hideText
-                data-test="btn-delete-questionnaire"
-                onClick={partial(onDeleteQuestionnaire, questionnaire.id)}
+      <>
+        <DeleteConfirmDialog
+          isOpen={showDeleteQuestionnaireDialog}
+          onClose={this.handleCloseDeleteQuestionnaireDialog}
+          onDelete={this.handleDeleteQuestionnaire}
+          title={questionnaire.title}
+          alertText="Questionnaire including all sections and questions be deleted."
```
Comment (hamishtaplin) [inline eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.js]:
```suggestion
          alertText="This questionnaire including all sections and questions will be deleted."
```

Comment (rmccar) [reply eq-author/src/App/QuestionnairesPage/QuestionnairesTable/Row.js]: Sorted

**Pull request merged**